### PR TITLE
Task workflow Model and Endpoints

### LINKED ISSUE #273: Task Workflow - Model & Endpoints
In order to get started with the CK Workspace tool, we'll need to create a new type of workflow called the `TaskWorkflow`.

Roughly, it should include the following fields since we're trying to follow this [scheme](https://docs.google.com/presentation/d/1HCbXmwMkj3ZgOKnBdI348iymSdOGt4a3/edit#slide=id.g1095379bf74_0_0):
```
export enum TaskAction {
  LIKE = 'LIKE',
  COMMENT = 'COMMENT',
  TAG = 'TAG',
}

class TaskWorkflowModel extends WorkflowModel {
  prompt: string; // 

  actions: TaskAction[]; // actions needed for this task
  assignedGroups: Group[]; // groups that are assigned this task

  ... add anything else that may seem fit
}
```

For the endpoints, we'll just need our basic CRUD abilities, for now atleast:
```
POST /api/workflows/task - Creates a new task workflow
POST /api/workflows/task/:id - Update an existing task workflow

GET /api/workflows/task/boards/:id - Get all task workflows for a specific board

DELETE /api/workflows/task/:id - Delete a specific task workflow

... add any other endpoints you think may be useful
```

### PR DESCRIPTION
**Pull request by zarifmahmud**
## Details
- Creates model and endpoints for task workflow

Closes #273

### CODE REVIEW

File: backend/src/api/workflows.ts [hunk 1]
```diff
@@ -75,4 +74,72 @@ router.delete("/distribution/:id", async (req, res) => {
...
+
+  const workflow: Partial<TaskWorkflowModel> = Object.assign(
+    {},
+    name === null ? null : { name },
+    active === null ? null : { active },
+    source === null ? null : { source },
+    destinations === null ? null : { destinations },
+    prompt === null ? null : { prompt },
+    requiredActions === null ? null : { requiredActions },
+    optionalActions === null ? null : { optionalActions },
```
Comment (Ammar-T) [inline backend/src/api/workflows.ts:106]: I think we can remove the `optionalActions` field for now. I know it's in the mockups but we can leave it out for now.

File: backend/src/models/Workflow.ts [hunk 2]
```diff
@@ -51,9 +64,25 @@ export class DistributionWorkflowModel extends WorkflowModel {
   public postsPerDestination!: number;
 }
 
+
+export class TaskWorkflowModel extends WorkflowModel {
+  @prop({ required: true})
+  public prompt!: string;  
+  public requiredActions!: TaskAction[]; 
+  public optionalActions!: TaskAction[]; // Can be empty
```
Comment (Ammar-T) [inline backend/src/models/Workflow.ts:72]:
```suggestion
```

**Comment by Ammar-T** [comment]
@zarifmahmud Updates on this?

**Comment by zarifmahmud** [comment]
I removed the optionalActions field, and added the new TaskAction class

File: backend/src/api/workflows.ts (hunk 1 above)
Comment (Ammar-T) [reply backend/src/api/workflows.ts:106]: @zarifmahmud Still need to remove this.

Comment (zarifmahmud) [reply backend/src/api/workflows.ts]: Ah sorry, didn't see that it was also in workflow.ts

**Review by Ammar-T — APPROVED** [review]
👍

**Comment by JoelWiebe** [comment]
@zarifmahmud @Ammar-T Is this ready to be merged?

**Comment by zarifmahmud** [comment]
Yep I think it is

**Comment by JoelWiebe** [comment]
@zarifmahmud If all looks good, feel free to merge then delete the branch!

**Pull request merged**